src/main/java/uk/gov/justice/probation/courtcasematcher/model/domain/Address.java

```diff
@@ -12,11 +12,89 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
 @Builder
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class Address {
+public class Address implements Comparable<Address> {
     private final String line1;
     private final String line2;
     private final String line3;
     private final String line4;
     private final String line5;
     private final String postcode;
+
+    @Override
+    public int compareTo(Address addressToCompare) {
+        if (this.getLine1() == null && addressToCompare.getLine1() == null) {
+            // pass
+        } else if (this.getLine1() == null) {
+            return -1;
+        } else if (addressToCompare.getLine1() == null) {
+            return 1;
+        } else {
+            int line1Comparison = this.getLine1().compareTo(addressToCompare.getLine1());
+            if (line1Comparison != 0) {
+                return line1Comparison < 0 ? -1 : 1;
```

Comment:
I think this could be simplified:
```suggestion
                return line1Comparison;
```